File: 1995/4/2-4.py
#!/usr/bin/env python

import sys

chk = [1,0,0,0,0,0,1]
for line in sys.stdin:
    q = map(int, list(line))
    recording = False
    for i in xrange(len(q) / 7):
        if q[i*7:(i+1)*7] == chk:
            recording = not recording
        elif recording:
            sys.stdout.write(chr(sum([x * 2 ** (6-i) for i,x in enumerate(q[i*7:(i+1)*7])])))
    sys.stdout.write('\n')

File: 1995/4/2-4.rb
#!/usr/bin/env ruby

chk = [1,0,0,0,0,0,1]
File.readlines('prob4.in').each do |line|
  line = line.strip.chars.collect { |c| c.to_i }
  recording = false
  line.each_slice(7) do |slice|
    if slice == chk
      recording = !recording
    elsif recording
      sum = 0
      slice.each_with_index { |a,j| sum += a * (2 ** (6 - j)) }
      print sum.chr
    end
  end
  puts
end
